packages/cspell-types/src/CSpellSettingsDef.ts

```diff
@@ -120,17 +132,39 @@ export interface FileSettings extends ExtendableSettings, CommandLineSettings {
 }
 
 export interface ExtendableSettings extends Settings {
-    /** Overrides to apply based upon the file path. */
+    /**
+     * Overrides are used to apply settings for specific files in your project.
+     *
+     * For example:
+     *
+     * ```javascript
+     * "overrides": [
+     *   // Force `*.hrr` and `*.crr` files to be treated as `cpp` files:
+     *   {
+     *     "filename": "**\/{*.hrr,*.crr}",
```

Comment:
Yeah, this part is confusing, sorry.
For this part, I needed to integrate the glob into a JSDoc comment so that it would get captured by the automatic-doc-generator, and escaping the globs was tricky.

See https://github.com/streetsidesoftware/cspell/pull/2800/commits/b3253422d3ea10a7d1603cc8712b7b099c43b532